internal/handlers/sqlite/msg_dropindexes.go

```diff
@@ -16,13 +16,234 @@ package sqlite
+		"nIndexesWas", int32(len(beforeDrop.Indexes)),
+	))
+
+	if dropAll {
+		replyDoc.Set("msg", "non-_id indexes dropped for collection")
+	}
+
+	replyDoc.Set(
+		"ok", float64(1),
+	)
+
+	var reply wire.OpMsg
+	must.NoError(reply.SetSections(wire.OpMsgSection{
+		Documents: []*types.Document{replyDoc},
+	}))
+
+	return &reply, nil
+}
+
+// processDropIndexOptions parses index doc and returns the list of indexes to delete
+// and true if a flag to drop all indexes except _id_ was set.
+func processDropIndexOptions(command string, v any, existing []backends.IndexInfo) ([]string, bool, error) { //nolint:lll // for readability
+	switch v := v.(type) {
+	case *types.Document:
+		// Index specification (key) is provided to drop a specific index.
+		var spec []backends.IndexKeyPair
+
+		spec, err := processIndexKey(command, v)
+		if err != nil {
+			return nil, false, lazyerrors.Error(err)
+		}
+
+		if len(spec) == 1 && spec[0].Field == "_id" && !spec[0].Descending {
+			return nil, false, commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrInvalidOptions, "cannot drop _id index", command,
+			)
+		}
+
+		for _, index := range existing {
+			for i, key := range index.Key {
+				if key.Field == spec[i].Field && key.Descending == spec[i].Descending {
+					return []string{index.Name}, false, nil
+				}
+			}
+		}
+
+		formattedSpec := make([]string, len(spec))
+
+		for i, key := range spec {
+			order := 1
+			if key.Descending {
+				order = -1
+			}
+
+			formattedSpec[i] = fmt.Sprintf("%s: %d", key.Field, order)
+		}
+
+		msg := fmt.Sprintf("can't find index with key: { %v }", strings.Join(formattedSpec, ", "))
+
+		return nil, false, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrIndexNotFound, msg, command)
+
+	case *types.Array:
+		// List of index names is provided to drop multiple indexes.
+		iter := v.Iterator()
+
+		defer iter.Close() // it's safe to defer here as the iterators reads everything
+
+		toDrop := make([]string, 0, v.Len())
+
+		for {
+			var val any
+
+			_, val, err := iter.Next()
+			if err != nil {
+				if errors.Is(err, iterator.ErrIteratorDone) {
+					return toDrop, false, nil
+				}
+
+				return nil, false, lazyerrors.Error(err)
+			}
+
+			index, ok := val.(string)
+			if !ok {
+				return nil, false, commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrTypeMismatch,
+					fmt.Sprintf(
+						"BSON field 'dropIndexes.index' is the wrong type '%s', expected types '[string, object]'",
+						commonparams.AliasFromType(v),
+					),
+					command,
+				)
+			}
+
+			var found bool
+
+			for _, existingIndex := range existing {
+				if index == existingIndex.Name {
+					toDrop = append(toDrop, index)
+					found = true
+
+					break
+				}
+			}
+
+			if !found {
+				return nil, false, commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrIndexNotFound,
+					fmt.Sprintf("index not found with name [%s]", index),
+					command,
+				)
+			}
+		}
```

Comment:
We want to return found indexes `toDrop` right? :thinking: 
```suggestion
		}
		return toDrop, false, nil
```

Reply:
That's done above, where we check for `ErrIteratorDone`

Reply:
Yes, I missed that :sweat_smile: